crates/uv/src/lib.rs

```diff
@@ -903,7 +843,74 @@ async fn run() -> Result<ExitStatus> {
     }
 }
 
-fn main() -> ExitCode {
+pub fn main<I, T>(args: I) -> ExitCode
+where
+    I: IntoIterator<Item = T>,
+    T: Into<OsString> + Clone,
+{
+    // `std::env::args` is not `Send` so we parse before passing to our runtime
+    // https://github.com/rust-lang/rust/pull/48005
+    let cli = match Cli::try_parse_from(args) {
```

Comment:
This moved without change from `run` above because it's not safe to send `std::env` values across threads depending on the platform.